pkg/config/app_config.go

```diff
@@ -244,17 +232,41 @@ func (c *AppConfig) GetTempDir() string {
 	return c.TempDir
 }
 
-func configFilePath(filename string) (string, error) {
-	folder, err := findOrCreateConfigDir()
-	if err != nil {
-		return "", err
+func findConfigFile(filename string) (exists bool, path string) {
+	if envConfigDir := os.Getenv("CONFIG_DIR"); envConfigDir != "" {
+		return true, filepath.Join(envConfigDir, filename)
+	}
+
+	legacyConfigPath, err := xdg.SearchConfigFile(filepath.Join("jesseduffield", "lazygit", filename))
+	if err == nil {
+		return true, legacyConfigPath
+	}
+
+	configFilepath, err := xdg.SearchConfigFile(filepath.Join("lazygit", filename))
+	if err == nil {
+		return true, configFilepath
 	}
 
-	return filepath.Join(folder, filename), nil
+	return false, filepath.Join(xdg.ConfigHome, "lazygit", filename)
```

Comment:
It's not clear to me what's changed here compared to the previous code: Is xdg.ConfigHome different to what we look at when we do xdg.SearchConfigFile? Could we add comments giving example folders for a linux machine for each step?

Reply:
> It's not clear to me what's changed here compared to the previous code

This function is extracted from ConfigDir and configDirForVendor, the search order wasn't changed, but this version doesn't create extra directories (`stateFilePath` needs to use this function without creating any dirs) 

> Is xdg.ConfigHome different to what we look at when we do xdg.SearchConfigFile

`xdg.SearchConfigFile` errors out if the file doesn't exist, and `xdg.ConfigFile` creates parent directories, which I don't want, so I went with xdg.ConfigHome

> Could we add comments giving example folders for a linux machine for each step?

i added some comments